### DISCUSSION TITLE
Modelling a binary/step response with PASTAS

### DISCUSSION THREAD

**Discussion by benne94** [pastas/pastas#610] [Q&A] [answered]
I am trying to establishing pumping Influences of multiple abstraction wells on nearby groundwater levels.  I am currently running the following workflow:

1. Load pumping well data for multiple abstraction wells around a tailings storage facility (TSF).
2. Load GWLs for multiple obs wells surrounding the TSF.
3. Loop through individual wells and run a PASTAS model with rainfall only, then add the various stress models for individual wells.
4. Assess the influence of the ABS well on predicting the time-series to infer connectivity.

The results thus far have shown signs of usefulness, there is no data pre-pumping so there's not much of a warmup for the model.  The case below shows an example where I think there's sufficient evidence to confirm the pumping influence despite fairly low R squared values of 0.8, where orange line is recharge only model and blue shows with pumping well. 

I am now looking to incorporate a binary response to represent the installation of a liner in 2018 which limits seepage rates.  Does anyone have any suggestions on how I may do this? I picture this as a sort of logistic regression curve from 0 to 1 and then allow the response function to vary to try and identify potential lag-times between the liner going on and groundwater levels dropping off.

![image](https://github.com/pastas/pastas/assets/83494177/9f774e88-d1c4-4eba-8575-7ec44bcdfb03)

Many thanks in advance!

**Comment by mbakker7** [comment]
Dear @benne94,
When they install a liner on the storage facility, that will effect all the response functions as it is a change in the system. An approach to change from one response function to another response function was developed in https://ngwa.onlinelibrary.wiley.com/doi/full/10.1111/gwat.12891. I think we had an experimental implementation of this approach in Pastas, but I am not sure. Hopefully someone else knows.

**Reply by dbrakenhoff** [reply, marked as answer]
[This notebook](https://pastas.readthedocs.io/en/latest/examples/changing_responses.html) shows how to implement a change in the response function. 

A change in response function can be implemented using `ps.ChangeModel` and takes a stress (e.g. pumping) and two response functions as its input and a time at which the change occurred. For example:

```python
import pastas as ps

# load data ...

# create model
ml = ps.Model(head)

# add changing response
rfunc1 = ps.Gamma()  # response function before change
rfunc2 = ps.Gamma()  # response function after change
cm = ps.ChangeModel(well_discharge, rfunc1, rfunc2, "well", tchange="2018-01-01")
ml.add_stressmodels(cm)

# solve model
ml.solve()
```
